### ISSUE TITLE
Plotting duplicated annotated line produces corrupt PS

### ISSUE THREAD

**Issue by joa-quim** [GenericMappingTools/gmt#3530]
From forum [post](https://forum.generic-mapping-tools.org/t/psxy-sq-stopped-working/571/2)

data ``t.dat``
```
> -L-100 -Sqd1.0i
-86.45235040920656 44.56074532391791
-87.15496366731526 44.63460538965384
-87.52203359612746 43.70375243465723
-87.57565650687522 42.81267315180862
-87.46872284091027 42.21696862267483
-87.29959207789203 41.98325754479477
-86.88164979096022 41.9742639278875
-86.56233691428744 42.29261909586023
-86.46591956240054 42.8760163962448
-86.67090781901327 43.41777121767043
-86.7207951116628 43.91664988596499
-86.45235040920656 44.56074532391791
> -L-100 -Sqd1.0i
-86.45235040920656 44.56074532391791
-87.15496366731526 44.63460538965384
-87.52203359612746 43.70375243465723
-87.57565650687522 42.81267315180862
-87.46872284091027 42.21696862267483
-87.29959207789203 41.98325754479477
-86.88164979096022 41.9742639278875
-86.56233691428744 42.29261909586023
-86.46591956240054 42.8760163962448
-86.67090781901327 43.41777121767043
-86.7207951116628 43.91664988596499
-86.45235040920656 44.56074532391791
```
command
```
gmt psxy t.dat -R-88.50/41.45/-84.56/46.23r -Jt-86.3855/1:2175000 -P -W0.50p,black -Sqd1.0i:+d+Lh > lixo.ps
```
Removing the duplicated segment makes it plot correctly.

**Comment by PaulWessel** [comment]
Related to the segment header.  Removing them and I get a plot with duplicated lines.  There may be some restrictions on what changes to -Sq can be done via segment headers.  WIll have to see in debug what actually happens to give me a SEGV.

**Comment by joa-quim** [comment]
Dumb VS2019 no longer systematically crashes in those case, which was extremely useful for it would lead directly to the crashing point.

**Comment by PaulWessel** [comment]
Scary that we must rely on Apple clang...

**Comment by joa-quim** [comment]
Yep, have to resort to Julia to when want to make use of that facility.

**Comment by PaulWessel** [comment]
Just thinking out aloud how **-S** in command line and segment header should work (I am not saying it works this way, but perhaps that is should):

1. If no **-S** is given on the command line then any **-S** in a segment header is completely ignored (you are basically selecting line or polygon mode).
2. If an unspecified symbol is given on the command line (**-S**) then the **-S** in  a segment header is fully parsed.
3. If a complete **-S** is given on the command line then any **-S** in a segment header should be ignored.

I am not sure about the third, but there should be a way to have either the command line (full syntax) or the segment header (**-S**) be in effect, no?  We usually let command line options override defaults, and I guess in this context, settings in the segment header constitute defaults (?).
